addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3267,46 +3267,59 @@ export class OdooEditor extends EventTarget {
+                        node.oShiftTab(0);
+                    } else {
+                        node = closestBlock(node).firstElementChild
+                        if (isEditorTab(node) && !getAdjacentPreviousSiblings(node, sibling => (
+                            sibling.nodeType !== Node.TEXT_NODE || !/^[\u200B\s]*$/.test(sibling.textContent)
+                        )).length) {
+                            let {anchorNode, anchorOffset, focusNode, focusOffset} = sel;
+                            const [updateAnchor, updateFocus] = [anchorNode === node.nextSibling, focusNode === node.nextSibling];
+                            let zwsRemoved = 0;
+                            while (node.nextSibling && node.nextSibling.nodeType === Node.TEXT_NODE && node.nextSibling.textContent.startsWith('\u200B')) {
+                                splitTextNode(node.nextSibling, 1, DIRECTIONS.LEFT);
+                                node.nextSibling.remove();
+                                zwsRemoved++;
+                            }
+                            if (updateAnchor || updateFocus) {
+                                setSelection(
+                                    updateAnchor ? node.nextSibling : anchorNode,
+                                    updateAnchor ? Math.max(0, anchorOffset - zwsRemoved) : anchorOffset,
+                                    updateFocus ? node.nextSibling : focusNode,
+                                    updateFocus ? Math.max(0, focusOffset - zwsRemoved) : focusOffset
+                                );
+                            }
+                            node.remove();
+                        }
                     }
-                    tab.remove();
                 });
                 this.historyStep();
-            } else if (sel.isCollapsed) {
+            } else if (sel.isCollapsed && !closestElement(sel.anchorNode, 'li')) {
                 this.execCommand('insert', parseHTML('<span class="oe-tabs" contenteditable="false">\u0009</span>\u200B'));
             } else {
                 const restore = preserveCursor(this.document);
                 const tab = parseHTML('<span class="oe-tabs" contenteditable="false">\u0009</span>\u200B');
-                new Set(getTraversedNodes(this.editable).map(node => closestBlock(node)))
-                    .forEach(block => block.prepend(tab.cloneNode(true)));
+                new Set(getTraversedNodes(this.editable).map(node => closestBlock(node, 'li') || closestBlock(node)))
+                    .forEach(block => {
+                        if (descendants(block).find(descendant => descendant.nodeName === 'LI')) {
+                            return;
+                        }
+                        if (block.nodeName === 'LI') {
+                            block.oTab(0);
+                        } else {
+                            block.prepend(tab.cloneNode(true));
+                        }
+                    });
```

Comment:
I do not understand. There looks to be a lot of changes here but I don't understand what was wrong with the current behavior of tab in lists ? It already indents/unindents the points and already does it in group if the range spans multiple bullet points so what was the goal here ? The commit message isn't of any help and there are no new tests regarding lists or I missed them.